ouroboros-consensus/src/Ouroboros/Consensus/TxSubmission.hs

```diff
@@ -32,12 +32,11 @@ localTxSubmissionServer tracer Mempool{addTxs} =
       recvMsgSubmitTx = \tx -> do
         traceWith tracer (condense tx)
         res <- addTxs [tx]
-        -- The 'addTxs' action returns the failing ones, whereas the protocol
-        -- returns a Maybe failure for a single tx, so we must convert here.
         case res of
-          [(_tx,rej)] -> return (Just (show rej), server)
-          _           -> return (Nothing,         server),
+          [(_tx, mbErr)] -> return (show <$> mbErr, server)
+          -- The output list of addTxs has the same length as the input list.
+          _              -> error "addTxs: unexpected result"
```

Comment:
This patch probably was intended to be in a separate PR. It is good but unrelated.

Reply:
Indeed, I'll make a separate PR for it.

Reply:
See #840